### ISSUE TITLE
Beats build run from the unified build process fails `sometime` on packetbeat

### ISSUE THREAD

**Issue by ph** [elastic/beats#24304]
When beats is build from the unified build process something it will fail to build packetbeat and will return the following error. Reckicking the build can fix the problem.

Few things to investigate:

- Is there something wrong with the libpcap artifact? We use a sha256 to validates the artifacts so It should download and validates the artifacts.
- Is there a race condition when installing the libpcap and compiling it.

```
13:39:40 >> golangCrossBuild: Building for windows/386
13:39:40 >> Building using: cmd='build/mage-linux-amd64 golangCrossBuild', env=[CC=i686-w64-mingw32-gcc, CXX=i686-w64-mingw32-g++, GOARCH=386, GOARM=, GOOS=windows, PLATFORM_ID=windows-386]
13:39:41 conflicts: 38 shift/reduce
13:39:50 configure: WARNING: using cross tools not prefixed with host triplet
13:39:50 ./configure: line 8365: ./conftest: cannot execute binary file: Exec format errorPlease post all questions and issues on https://discuss.elastic.co/c/beats
before opening a Github Issue. Your questions will reach a wider audience there,
and if we confirm that there is a bug, then you can open a new issue.

For security vulnerabilities please only send reports to security@elastic.co.
See https://www.elastic.co/community/security for more information.

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: 
- Operating System: 
- Discuss Forum URL:
- Steps to Reproduce: 

13:39:51 conflicts: 38 shift/reduce
13:39:52 grammar.y: warning: 38 shift/reduce conflicts [-Wconflicts-sr]
13:39:55 # crypto/x509
13:40:32 ld: warning: object file (/tmp/go-build624014029/b101/_cgo_main.o) was built for newer OSX version (10.11) than being linked (10.10)
13:40:32 ld: warning: object file (/tmp/go-build624014029/b101/_x001.o) was built for newer OSX version (10.11) than being linked (10.10)
13:40:32 ld: warning: object file (/tmp/go-build624014029/b101/_x002.o) was built for newer OSX version (10.11) than being linked (10.10)
13:40:32 # github.com/elastic/beats/v7/x-pack/packetbeat
13:43:08 asm: open ./dlbh.s: no such file or directory
13:43:08 Error: running "go build -o build/golang-crossbuild/packetbeat-darwin-amd64 -ldflags -s -X github.com/elastic/beats/v7/libbeat/version.buildTime=2021-03-02T18:39:39Z -X github.com/elastic/beats/v7/libbeat/version.commit=9ddc25c174cd7a187296f1cdb25a38a54c07dd20" failed with exit code 2
13:43:09 Error: failed building for darwin/amd64: exit status 2
13:43:09 failed building for darwin/amd64: exit status 2
13:43:09 package ran for 3m35.511455367s
13:43:09 Error: running "docker run --env EXEC_UID=1161 --env EXEC_GID=1162 -v /var/lib/jenkins/pkg/mod:/go/pkg/mod:ro --rm --env GOFLAGS=-mod=readonly --env MAGEFILE_VERBOSE= --env MAGEFILE_TIMEOUT= --env SNAPSHOT=false --env DEV=false -v /var/lib/jenkins/workspace/elastic+unified-release+master+staging-standalone-build-beats-7.12/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/beats:/go/src/github.com/elastic/beats -w /go/src/github.com/elastic/beats/x-pack/packetbeat docker.elastic.co/beats-dev/golang-crossbuild:1.15.8-darwin --build-cmd build/mage-linux-amd64 golangCrossBuild -p darwin/amd64" failed with exit code 1
13:43:09 make[2]: *** [../../dev-tools/make/mage.mk:41: release] Error 1
13:43:09 make[2]: Leaving directory '/var/lib/jenkins/workspace/elastic+unified-release+master+staging-standalone-build-beats-7.12/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/beats/x-pack/packetbeat'
13:43:09 make[1]: *** [Makefile:201: release] Error 1
13:43:09 make: *** [Makefile:214: release-manager-release] Error 2
13:43:09 make[1]: Leaving directory '/var/lib/jenkins/workspace/elastic+unified-release+master+staging-standalone-build-beats-7.12/cd/release/release-manager/build/projects/gopath/src/github.com/elastic/beats'
13:43:09 
13:43:09 > Task :localBuildBeatsRelease-local FAILED
13:43:09 :localBuildBeatsRelease-local (Thread[Execution worker for ':',5,main]) completed. Took 1 hrs 32 mins 5.858 secs.
13:43:09 
13:43:09 FAILURE: Build failed with an exception.
13:43:09 
13:43:09 * What went wrong:
13:43:09 Execution failed for task ':localBuildBeatsRelease-local'.
```

**Comment by ph** [comment]
@urso and @andresrc we should fix this in the iteration. The build fails randomly and it hurts BC and the snapshots.

**Comment by ph** [comment]
@andrewkroh This might be a long shot, but do you have a theory about what could be causing this? I am not aware of major build changes that could have affected packetbeat. This code has been rock solid for quite some time even years.

**Comment by andrewkroh** [comment]
> ./configure: line 8365: ./conftest: cannot execute binary file: Exec format error

libpcap gets compiled inside the container at a location that's part of the container FS to avoid any conflicts with other parallel builds. So this one surprises me. I'm not aware of any recent Packetbeat changes. Maybe check if anything in dev-tools mage changed recently.

> asm: open ./dlbh.s: no such file or directory

This sounds similar to the comment at https://github.com/elastic/beats/blob/28d48145bbfe2131d48ed5f6756853070d7d5492/packetbeat/magefile.go#L66-L69 which at the time only affected Windows AFAICT, but perhaps something change in Go where darwin needs to be isolated too.

**Comment by andrewkroh** [comment]
Another option is to remove the libpcap from Beats and put it into golang-crossbuild. This would eliminate where it runs `./configure` in the Packetbeat build.

**Comment by ph** [comment]
closed by #25340
